composer/trainer/ddp.py

```diff
@@ -82,16 +85,19 @@ def __init__(self,
                  num_nodes: int,
                  backend: str,
                  fork_rank_0: bool,
+                 timeout: float,
                  find_unused_parameters: bool = False):
+        self.hparams = DDPHparams(
```

Comment:
Are we for sure using this pattern everywhere? There are some places (ex. `trainer.py`) where we do the opposite (call init from `create_from_hparams`).

Reply:
trainer.py is the exception to this rule, just because there is so much logic in there that really belongs in the trainer.py and not the hparams. But otherwise, `create_from_hparams` should call init, which then recreates the hparams.